lib/clarivate/links_client.rb

```diff
@@ -0,0 +1,65 @@
+module Clarivate
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+    attr_reader :username, :password, :host
+
+    def initialize(username: nil, password: nil, host: 'https://ws.isiknowledge.com')
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # limited to 50 ids
+    # @param [Array<String>] ids
+    # @return [Hash<String => Hash>]
+    def links(ids)
+      raise ArgumentError, "1-50 ids required" unless ids.present? && ids.count <= 50
+      response = connection.send(:post) do |req|
+        req.path = '/cps/xrpc'
+        req.body = request_body(ids.uniq)
+      end
+      ng = Nokogiri::XML(response.body) { |config| config.strict.noblanks }.remove_namespaces!
+      pairs = ng.xpath('response/fn/map/map').map do |node|
+        [node.attr('name').split('_').last, vals_to_hash(node.xpath('//val'))]
+      end
+      pairs.to_h
+    end
+
+    private
+
+      # @return [Faraday::Connection]
+      def connection
+        @connection ||= Faraday.new(url: host) do |faraday|
+          faraday.use Faraday::Response::RaiseError
+          faraday.adapter :httpclient
+        end
+      end
+
+      # @return [ActionView::Base] used to render as though in an rails controller
+      def renderer
+        @renderer ||= ActionView::Base.new(ActionController::Base.view_paths, {})
+      end
+
+      def request_body(ids)
+        renderer.render(
+          file: 'pages/clarivate_links.xml',
+          layout: false,
+          locals: {
+            client: self,
+            return_fields: %w(ut doi pmid),
+            cites: ids
+          }
```

Comment:
Do we need the `ut` in the `return_fields`?  Can we assume that their response map is always reliable?  Can we assume that there can only be one `ut` and if we have it in the hash-key, that's enough?

Reply:
No reason not to have it.  Also makes sense if we start querying by different input types.

Reply:
In #325 the fields are an input parameter to the query, rather than have them fixed.